### PR TITLE
Add contributing.md

### PR DESCRIPTION
**Pull request by sandre35**
This is a first step to solve the following issue : https://github.com/tournesol-app/tournesol/issues/207

### CODE REVIEW

File: CONTRIBUTING.md
Comment (GresilleSiffle) [inline CONTRIBUTING.md:24]: How about using `Moreover` instead of `However`? As the second sentence is not contradictory with the first, I think `Moreover` would be more appropriate.

```suggestion
Moreover the command `pytest --html=report.html –self-contained-html` enables to create a clean html report document with details of logs for each test.
```

File: CONTRIBUTING.md
Comment (GresilleSiffle) [inline CONTRIBUTING.md:28]: We can be more precise by naming what is exactly the .yaml file, and what are the typescript files.

Also, I think we should explain when a developper need to run yarn run update-schema, what do you think? Here is a suggestion.

```suggestion
When the back end API is updated, the front end must re-generate its local client to match the new API schema.

**step 1** While the back end is running, from the root of the front end folder run the command `yarn run update-schema`. This will update front end representation the back end OpenAPI schema.

**step 2** The re-generate the service files by running `scripts/generate-services-from-openapi.sh`
```

File: README.md
Comment (GresilleSiffle) [inline README.md:30]: I propose to openly make the `CONTRIBUTING.md` file name appear here.

```suggestion
See the [CONTRIBUTING.md](./CONTRIBUTING.md) file.
```

File: README.md
Comment (GresilleSiffle) [inline README.md:31]: @amatissart @lfaucon 

Should we update or hide this invitation to check the Wiki?

The `CONTRIBUTING.md` will contain a lot of details too, so this raises the question why a user would click on the Wiki link instead of the Wiki? What content will be in the Wiki and not in the CONTRIBUTING.md?

I think we could be more precise by saying "See the Wiki if you want to XXX ". This will avoid having two distinct sources of truth referenced in the same document.

For instance, we could decide that all contribution instructions go to the CONTRIBUTING.md file. Or, all contribution instructions go to the CONTRIBUTING.md file expect those explaining how to compare videos, or how to be involved in social activities like brainstorming or so.

**Review by GresilleSiffle — CHANGES_REQUESTED** [review]
Thanks @sandre35 !

I've suggested minor typo fixes, and some rewording. You'll be able to integrate them quickly by simply clicking on the `Commit suggestion` button under each suggestion. I think it's also a good idea that @amatissart and @lfaucon have a look also as the document quotes us all, and is rather small.

On my side, I'm OK to merge the document as is, after the suggestion have been addressed. It will be a good starting point to flatten our other contribution processes ( creating issues, feature requests, etc.).

File: CONTRIBUTING.md
Comment (amatissart) [reply CONTRIBUTING.md:7]: GitHub also allow to define teams under an organization. So creating a team `@tournesol-app/maintainers` could make sense.

Only organization owners can create a team though, on https://github.com/orgs/tournesol-app/new-team

File: README.md
Comment (amatissart) [reply README.md:31]: And most of the content on the Wiki page "Contribute to Tournesol" is outdated. So I would agree to remove it for now, until the need for this wiki page is clarified.

File: CONTRIBUTING.md
Comment (sandre35) [reply CONTRIBUTING.md:14]: Yes, this is what I thought when I wrote it

File: CONTRIBUTING.md
Comment (sandre35) [reply CONTRIBUTING.md:20]: I tried to express the idea that creating end-to-end test is not something that is done regularly.

File: CONTRIBUTING.md
Comment (sandre35) [reply CONTRIBUTING.md:8]: @amatissart As i understand, you first need to create the team. Then I can mention directly this team inside the document.
If I am right, let me know when the team is created

File: README.md
Comment (sandre35) [reply README.md]: I have removed the link to the wiki page

File: CONTRIBUTING.md
Comment (lfaucon) [reply CONTRIBUTING.md:8]: We should have a "maintainer" team now (Gresille should have a pending invite and I'll try to add amatissart to it too as soon as I figure out how to free one extra seat in the organization or choose to pay for it)

https://github.com/orgs/tournesol-app/teams/maintainers

**Review by lfaucon — APPROVED** [review]
Approved from my side, once other reviewer's comments are addressed. Thanks for the docs contribution 📖  👍

**Comment by GresilleSiffle** [comment]
Hello @sandre35 

With the creation of the team @tournesol-app/maintainers , we can replace the all individual names with it.

After this, it will be ready to be merged.

**Comment by sandre35** [comment]
On my side, I can not access the page provided with this link : [@tournesol-app/maintainers](https://github.com/orgs/tournesol-app/teams/maintainers)
Maybe only maintainers can access this page?

**Comment by lfaucon** [comment]
I'll try looking into how to make teams public, or maybe @GresilleSiffle knows already?

**Comment by GresilleSiffle** [comment]
I don't know well how team works, but it seems that it's not required for a user to have access to the team space to be able to mention it.

If I understood correctly the team space is private, and allows us to have private discussions, but mentioning should be available for all GitHub users. 

Can you try to mention the maintainers @sandre35 ?

**Comment by lfaucon** [comment]
@tournesol-app/maintainers 👋 testing tagging maintainers

**Comment by sandre35** [comment]
Okay I see.
But if a new comer wants to talk to someone that is part of maintainers, he can't see is part of the team.
When asking for review he can not assign directly thé tag maintainers. The only to do it is by mentionning it in one comment on the PR

**Comment by lfaucon** [comment]
I think it is good enough to tag maintainers in a comment, then the maintainers will assign themselves for review as they prefer.

**Comment by lfaucon** [comment]
Screenshot 2022-01-24 at 19 15 31
When hovering above the teams tag, I can see who is in. Can you see it too @sandre35 ?

**Comment by sandre35** [comment]
> Screenshot 2022-01-24 at 19 15 31
> 
> When hovering above the teams tag, I can see who is in. Can you see it too @sandre35 ?

No, when I hover it nothing appears. Only the url on the bottom of the browser appears

**Comment by lfaucon** [comment]
Let's merge this with the explicit names. No problem. It will not change often enough that this would cost us

**Comment by GresilleSiffle** [comment]
I'm OK to merge it as is for the moment :+1: 

We can still upgrade this file step by step.

Also, removing the markdown link to the maintainers team seems good, to prevent users to click on it and be confused by the access denied.

**Pull request merged**
